vividus/src/main/java/org/vividus/results/CollectingStatisticsStoryReporter.java

```diff
@@ -104,9 +108,19 @@ public Optional<List<Failure>> getFailures()
     }
 
     @Override
-    public Optional<Status> getRunStatus()
+    public int getStatusCode()
     {
-        return status.get();
+        Optional<Status> runStatus = status.get();
+        if (runStatus.isEmpty())
+        {
+            return RUN_FAILED_EXIT_CODE;
+        }
+        switch (runStatus.get())
+        {
+            case PASSED:            return RUN_PASSED_EXIT_CODE;
+            case KNOWN_ISSUES_ONLY: return RUN_KNOWN_ISSUES_EXIT_CODE;
+            default:                return RUN_FAILED_EXIT_CODE;
+        }
```

Comment:
```suggestion
       return status.get().map(s -> {
          switch (s)
          {
              case PASSED:            return RUN_PASSED_EXIT_CODE;
              case KNOWN_ISSUES_ONLY: return RUN_KNOWN_ISSUES_EXIT_CODE;
              default:                return RUN_FAILED_EXIT_CODE;
          }
        }).orElse(RUN_FAILED_EXIT_CODE)
     
```

Reply:
prefer initial variant as the switch in map doesn't look good (just my opinion)